src/toil_scripts/batch_alignment/bwa_alignment.py

```diff
@@ -1,223 +1,254 @@
+        ann: s3://cgl-pipeline-inputs/alignment/hg19.fa.ann
+
+        # Optional: Reference fasta file (bwt) -- If not present will be generated
+        bwt: s3://cgl-pipeline-inputs/alignment/hg19.fa.bwt
+
+        # Optional: Reference fasta file (pac) -- If not present will be generated
+        pac: s3://cgl-pipeline-inputs/alignment/hg19.fa.pac
+
+        # Optional: Reference fasta file (sa) -- If not present will be generated
+        sa: s3://cgl-pipeline-inputs/alignment/hg19.fa.sa
+
+        # Optional: Reference fasta file (fai) -- If not present will be generated
+        fai: s3://cgl-pipeline-inputs/alignment/hg19.fa.fai
+
+        # Optional: (string) Path to Key File for SSE-C Encryption
+        ssec:
+
+        # Optional: Use instead of library, program_unit, and platform.
+        rg-line:
+
+        # Optional: Alternate file for reference build (alt). Necessary for alt aware alignment
+        alt:
+
+        # Optional: If true, runs the pipeline in mock mode, generating a fake output bam
+        mock-mode:
+
+        # Optional: Optional suffix to add to sample output
+        suffix:
     """[1:])
 
+
 def generate_manifest():
     return textwrap.dedent("""
-        #   Edit this manifest to include information pertaining to each sample to be run.
-        #   There are 3 tab-separated columns: UUID, URL1, URL2
+        #   Edit this manifest to include information for each sample to be run.
+        #   Lines beginning with # are ignored.
+        #
+        #   There are 2 or 3 tab-separated columns: UUID, 1st FASTQ URL, 2nd FASTQ URL (if paired)
+        #   If a sample is paired end: UUID    URL1    URL2
+        #   If a sample is single-ended: UUID    URL
         #
-        #   UUID        This should be a unique identifier for the sample to be processed
-        #   URL1/URL2         A URL (http://, ftp://, file://, s3://, gnos://) pointing to the paired sample fastq files
+        #   UUID            This should be a unique identifier for the sample.
+        #   URL1/URL2       A URL (http://, ftp://, file://, s3://, gnos://) pointing to the sample fastq files.
         #
-        #   Example below. Lines beginning with # are ignored.
+        #   Examples below:
         #
-        #   UUID_1    file:///path/to/R1.fq    file:///path/to/R2.fq
+        #   Paired_UUID    file:///path/to/R1.fq.gz    file:///path/to/R2.fq.gz
+        #   Unpaired_UUID    file:///path/to/unpaired.fq.gz
         #
-        #   Place your samples below, one per line.
+        #   Place your samples below, one sample per line.
         """[1:])
 
+
+def parse_manifest(manifest_path):
+    """
+    Parse manifest file
+
+    :param str manifest_path: Path to manifest file
+    :return: samples
+    :rtype: list[str, list]
+    """
+    samples = []
+    with open(manifest_path, 'r') as f:
+        for line in f.readlines():
+            if not line.isspace() and not line.startswith('#'):
```

Comment:
I could switch to `csv`, but since the manifest has comments it wouldn't be much cleaner:

```
with open(manifest_path, 'r') as tsv:
    for line in csv.reader(tsv, delimiter='\t'):
        if line and not line[0].startswith('#'):
....
````

I'd save one line by skipping the strip and split on tab.